File: app/src/main/java/com/imangazalievm/bubbble/di/UserProfilePresenterComponent.java
package com.imangazalievm.bubbble.di;

import com.imangazalievm.bubbble.di.modules.UserProfilePresenterModule;
import com.imangazalievm.bubbble.di.scopes.Presenter;
import com.imangazalievm.bubbble.presentation.mvp.presenters.UserFollowersPresenter;
import com.imangazalievm.bubbble.presentation.mvp.presenters.UserProfilePresenter;

import dagger.Component;

@Presenter
@Component(dependencies = ApplicationComponent.class, modules = UserProfilePresenterModule.class)
public interface UserProfilePresenterComponent {

    UserProfilePresenter getPresenter();

}

File: app/src/main/java/com/imangazalievm/bubbble/di/userprofile/UserProfilePresenterComponent.kt
package com.imangazalievm.bubbble.di.userprofile

import com.imangazalievm.bubbble.di.global.scopes.Presenter
import com.imangazalievm.bubbble.di.global.ApplicationComponent
import com.imangazalievm.bubbble.di.userprofile.UserProfilePresenterModule
import com.imangazalievm.bubbble.presentation.mvp.userprofile.UserProfilePresenter
import dagger.Component

@Presenter
@Component(
    dependencies = [ApplicationComponent::class],
    modules = [UserProfilePresenterModule::class]
)
interface UserProfilePresenterComponent {

    fun getPresenter(): UserProfilePresenter

}